minimal-kernel/Dockerfile

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Jupyter Development Team.
+FROM debian:jessie
+
+MAINTAINER Jupyter Project <jupyter@googlegroups.com>
+
+USER root
+
+# Install all OS dependencies for fully functional notebook server
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+    wget \
+    build-essential \
+    python3-dev \
+    python3-pip \
+    ca-certificates \
+    bzip2 \
+    unzip \
+    libsm6 \
+    python3-zmq \
+    sudo \
+    && apt-get clean
+
+# Install Tini
+RUN wget --quiet https://github.com/krallin/tini/releases/download/v0.6.0/tini && \
+    echo "d5ed732199c36a1189320e6c4859f0169e950692f451c03e7854243b95f4234b *tini" | sha256sum -c - && \
+    mv tini /usr/local/bin/tini && \
+    chmod +x /usr/local/bin/tini
+
+# Configure environment
+ENV SHELL /bin/bash
+ENV NB_USER jovyan
+ENV NB_UID 1000
+
+# Create jovyan user with UID=1000 and in the 'users' group
+# Grant ownership over the home dir, but stick the group as root.
+RUN useradd -m -s /bin/bash -N -u $NB_UID $NB_USER && \
+    mkdir /home/$NB_USER/work && \
+    mkdir /home/$NB_USER/.jupyter && \
+    mkdir /home/$NB_USER/.local && \
+    chown -R $NB_USER:users /home/$NB_USER
+
+# Install Kernel Gateway
+RUN pip3 install jupyter_kernel_gateway==0.2.0
+
+# Configure container startup
+USER $NB_USER
+EXPOSE 8888
+WORKDIR /home/$NB_USER/work
+ENTRYPOINT ["tini", "--"]
+CMD ["/usr/local/bin/start-notebook.sh"]
```

Comment:
Don't want to run start-notebook.sh. Entrypoint should be something like the one over the kernel gateway demos like `tini -- jupyter kernel_gateway` and the CMD is `--ip=0.0.0.0`